sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -276,41 +277,54 @@ def test_piecewise_simplify():
         Piecewise((1 + 1/x**2, Eq(x, 0)), ((-1)**(x + 1), True))
 
 
-def test_piecewise_solve():
+def test_piecewise_solveset():
+    x = Symbol('x', real=True)
     abs2 = Piecewise((-x, x <= 0), (x, x > 0))
     f = abs2.subs(x, x - 2)
+
     assert solve(f, x) == [2]
+    assert solveset(f, x) == FiniteSet(2)
+
     assert solve(f - 1, x) == [1, 3]
+    assert solveset(f - 1, x) == FiniteSet(1, 3)
 
     f = Piecewise(((x - 2)**2, x >= 0), (1, True))
     assert solve(f, x) == [2]
+    assert solveset(f, x) == FiniteSet(2)
 
     g = Piecewise(((x - 5)**5, x >= 4), (f, True))
     assert solve(g, x) == [2, 5]
+    assert solveset(g, x) == FiniteSet(2, 5)
 
     g = Piecewise(((x - 5)**5, x >= 4), (f, x < 4))
     assert solve(g, x) == [2, 5]
+    assert solveset(g, x) == FiniteSet(2, 5)
 
     g = Piecewise(((x - 5)**5, x >= 2), (f, x < 2))
     assert solve(g, x) == [5]
+    assert solveset(g, x) == FiniteSet(5)
 
     g = Piecewise(((x - 5)**5, x >= 2), (f, True))
     assert solve(g, x) == [5]
+    assert solveset(g, x) == FiniteSet(5)
 
     g = Piecewise(((x - 5)**5, x >= 2), (f, True), (10, False))
     assert solve(g, x) == [5]
+    assert solveset(g, x) == FiniteSet(5)
 
     g = Piecewise(((x - 5)**5, x >= 2),
                   (-x + 2, x - 2 <= 0), (x - 2, x - 2 > 0))
     assert solve(g, x) == [5]
+    assert solveset(g, x) == FiniteSet(5)
 
 # See issue 4352 (enhance the solver to handle inequalities).
 
 
 @XFAIL
 def test_piecewise_solve2():
+    # issue 9742
     f = Piecewise(((x - 2)**2, x >= 0), (0, True))
-    assert solve(f, x) == [2, Interval(0, oo, True, True)]
+    assert solveset(f, x) == Union(FiniteSet(2), Interval(-oo, 0, True, True))
```

Comment:
I did verify tests, rather than simply copying them, the previous solution was wrong, so I changed it, the solution to the following equation:

```
In [7]: f
Out[7]: 
⎧       2           
⎪(x - 2)   for x ≥ 0
⎨                   
⎪   0      otherwise
⎩    
```
```python
if x >= 0 : solution = {2}
if x < 0 : solution = Interval(-oo, 0)
```